src/hydamo/hydamo/code_utils.py

```diff
@@ -0,0 +1,125 @@
+"""Utilities for generating unique codes."""
+from pathlib import Path
+from typing import Union
+
+import pandas as pd
+from pandas import DataFrame
+
+CODES_CSV = Path(__file__).parent.joinpath("data", "codes.csv")
+CODES_DF = None
+
+WBH_CODE_TEMPLATE = "NL.WBHCODE.{wbh_code}.{code}"
+BGT_CODE_TEMPLATE = "NL.BGTCODE.{bgt_code}.{code}"
+
+
+def get_codes_df():
+    """Get (and set) the global CODES_DF.
+
+    Returns
+    -------
+    CODES_DF : DataFrame
+
+    """
+    global CODES_DF
+
+    # read BGT_CSV if CODES_DF is None
+    if CODES_DF is None:
+        CODES_DF = pd.read_csv(CODES_CSV)
+        CODES_DF.set_index(CODES_DF.name.str.lower(), inplace=True)
+        CODES_DF["wbh_code"] = CODES_DF.wbh_code.apply(
+            lambda x: f"{int(x):02}" if not pd.isna(x) else None
+        )
+    return CODES_DF.copy()
+
+
+def bgt_code_exists(bgt_code: str):
+    """Check if bgt_code exists in CODES_DF"""
+    codes_df = get_codes_df()
+    return bgt_code in codes_df.bgt_code.to_numpy()
+
+
+def wbh_code_exists(wbh_code):
+    """Check if wbh_code exists in CODES_DF"""
+    codes_df = get_codes_df()
+    return wbh_code in codes_df.wbh_code.to_numpy()
+
+
+def bgt_to_wbh_code(bgt_code):
+    """Convert bgt_code to wbh_code if bgt_code exists"""
+    if bgt_code_exists(bgt_code):
+        codes_df = get_codes_df()
+        return (
+            codes_df.reset_index(drop=True).set_index("bgt_code").loc[bgt_code].wbh_code
+        )
+
+
+def find_codes(
+    organization: str,
+    administration_category: Union[str, None] = None,
+    to_dict: bool = True,
+):
```

Comment:
These type annotations really make it easier to read. Could you also annotate the return type, so I know what comes back?

Reply:
enhanced it in #26